src/components/timeline.js

```diff
@@ -58,19 +64,19 @@ function Timeline({ length, parts, selectedPartIndex, onPartAdd, onPartDelete, o
 
     setDragState(prev => ({ ...prev, marker }));
 
-    const newStartTimecode = Math.round(dragState.marker.left / timelineState.msToPxRatio);
-    onStartTimecodeChange(dragState.marker.key, newStartTimecode);
-  }, [dragState, timelineState, onStartTimecodeChange]);
+    const newStartPosition = Math.max(0, Math.min(1, dragState.marker.left / timelineState.currentTimelineWidth));
+    onStartPositionChange(dragState.marker.key, newStartPosition);
+  }, [dragState, timelineState, onStartPositionChange]);
 
   const handleSegmentsBarClick = () => {
-    if (isTouchDevice()) {
+    if (!timelineState.currentTimelineWidth || isTouchDevice()) {
       return;
     }
 
     if (newMarkerState?.isInBounds) {
-      const startTimecode = Math.round(newMarkerState.left / timelineState.msToPxRatio);
+      const startPosition = Math.max(0, Math.min(1, newMarkerState.left / timelineState.currentTimelineWidth));
```

Comment:
Maybe this `Math.max(0, Math.min(1, <position>, <limit>))` could be extracted as a well named helper method, to make the code easier to grasp. Something like `getValueInUnitInterval` maybe, as in mathematics this [0,1] range is apparently called so.

Reply:
Good idea! I went for `ensureValidMediaPosition` so I could put it into `media-utils` (I couldn't find a good place otherwise).